cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/ExecutionSparkContext.java

```diff
@@ -246,18 +230,16 @@ public ExecutionSparkContext(ApplicationSpecification appSpec,
           }
           throw t;
         }
-        if (dataset instanceof DatasetOutputCommitter) {
-          ((DatasetOutputCommitter) dataset).onSuccess();
-        }
-        return;
+      } catch (ClassNotFoundException e) {
+        throw new DatasetInstantiationException(String.format(
+          "Cannot load input format class %s provided by dataset '%s'", outputFormatName, datasetName), e);
+      } catch (ClassCastException e) {
+        throw new DatasetInstantiationException(String.format(
+          "Input format class %s provided by dataset '%s' is not an input format", outputFormatName, datasetName), e);
```

Comment:
exception message seems wrong.
It's not an input format class, but rather a output format name, right?
Same with the message 3 lines above it.